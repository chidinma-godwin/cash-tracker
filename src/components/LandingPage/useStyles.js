import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  intro: {
    background: theme.palette.primary.main,
    position: 'relative',
    height: '80vh',
    padding: '3em',
    paddingTop: '2em',
  },
  buttonWrapper: {
    width: 'fit-content',
    marginLeft: 'auto',
    marginBottom: '1em',
  },
  buttonStyles: {
    fontSize: '1.5rem',
    color: '#fff',
    padding: '10px 20px',
    display: 'inline-block',
    position: 'relative',
    textDecoration: 'none',
    overflow: 'hidden',
    cursor: 'pointer',
    transition: '0.2s',
    borderRadius: '50px',

    '&:hover': {
      background: theme.palette.secondary.main,
      boxShadow: '0 0 10px #0ee623, 0 0 40px #0ee623, 0 0 80px #0ee623',
      transitionDelay: '0.5s',
    },

    '& span': {
      position: 'absolute',
      display: 'block',
      opacity: 0,
    },

    '& span:nth-child(1)': {
      top: 0,
      left: '-100%',
      height: '4px',
      width: '100%',
      background: `linear-gradient(90deg, transparent, ${theme.palette.secondary.main})`,
    },
    '&:hover span:nth-child(1)': {
      left: '100%',
      opacity: 1,
      transition: '0.5s',
    },
    '& span:nth-child(3)': {
      bottom: 0,
      right: '-100%',
      height: '4px',
      width: '100%',
      background: `linear-gradient(270deg, transparent, ${theme.palette.secondary.main})`,
    },
    '&:hover span:nth-child(3)': {
      right: '100%',
      opacity: 1,
      transition: '0.5s',
      transitionDelay: '0.3s',
    },
    '& span:nth-child(2)': {
      top: '-100%',
      right: 0,
      width: '4px',
      height: '100%',
      background: `linear-gradient(180deg, transparent, ${theme.palette.secondary.main})`,
    },
    '&:hover span:nth-child(2)': {
      top: '100%',
      opacity: 1,
      transition: '0.5s',
      transitionDelay: '0.2s',
    },
    '& span:nth-child(4)': {
      bottom: '-100%',
      left: 0,
      width: '4px',
      height: '100%',
      background: `linear-gradient(180deg, transparent, ${theme.palette.secondary.main})`,
    },
    '&:hover span:nth-child(4)': {
      bottom: '100%',
      opacity: 1,
      transition: '0.5s',
      transitionDelay: '0.4s',
    },
  },
  subSection: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    minHeight: '20vh',
    padding: '0 3em',
    background: theme.palette.secondary.main,
  },
  glowWrapper: {
    display: 'inline-flex',
    flexWrap: 'wrap',
  },
  glowText: {
    padding: 0,
    margin: 0,
    marginRight: '0.3em',
    color: '#000',
    fontSize: '6em',
    height: 'fit-content',
    letterSpacing: '4px',
    fontWeight: 'bolder',
    [theme.breakpoints.down('sm')]: {
      fontSize: '3rem',
    },
    '& span': {
      display: 'table-cell',
      margin: 0,
      padding: 0,
      color: '#fff',
      textShadow: '0 0 10px #fff, 0 0 40px #fff, 0 0 80px #fff',
    },
  },
  curve: {
    position: 'absolute',
    width: '100%',
    bottom: 0,
    left: 0,

    '& path': {
      fill: theme.palette.secondary.main,
    },
  },
  subtitle: {
    fontSize: '2rem',
    letterSpacing: '3px',
    paddingTop: '1rem',
    color: 'lavenderblush',
    [theme.breakpoints.down('sm')]: {
      fontSize: '1.5rem',
    },
  },
}));

export default useStyles;
