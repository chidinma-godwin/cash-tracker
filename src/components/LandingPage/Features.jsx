// CashTracker imports
import useStyles from './useStyles';

export default function Features() {
  const styles = useStyles();

  return (
    <section className={styles.subSection}>
      <h2>FEATURES</h2>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ducimus ipsum
        quas eum saepe molestiae numquam dolores autem facilis! Veniam harum nam
        dolore cumque, sed illum eligendi incidunt unde earum modi.
      </p>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Molestiae
        ullam quia animi voluptatem autem, temporibus dolore, pariatur ipsam
        reprehenderit sapiente numquam nobis molestias asperiores magnam culpa
        ratione deserunt id dignissimos.
      </p>
    </section>
  );
}
