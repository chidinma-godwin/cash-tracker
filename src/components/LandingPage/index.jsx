import Typography from '@material-ui/core/Typography';
import { useRouter } from 'next/router';

// CashTracker imports
import useStyles from './useStyles';

const LandingPage = () => {
  const styles = useStyles();
  const router = useRouter();

  const handleNavigate = location => {
    router.push('/account/[location]', `/account/${location}`);
  };

  return (
    <section className={styles.intro}>
      <div className={styles.buttonWrapper}>
        <a
          data-testid='login'
          className={styles.buttonStyles}
          onClick={() => handleNavigate('login')}
          onKeyDown={() => handleNavigate('login')}
          role='button'
          tabIndex={0}
        >
          <span />
          <span />
          <span />
          <span />
          Login
        </a>
        <a
          data-testid='sign-up'
          className={styles.buttonStyles}
          onClick={() => handleNavigate('signup')}
          onKeyDown={() => handleNavigate('signup')}
          role='button'
          tabIndex={0}
        >
          <span />
          <span />
          <span />
          <span />
          Sign Up
        </a>
      </div>

      <div className={styles.glowWrapper}>
        <Typography variant='h2' className={styles.glowText}>
          <span>C</span>
          <span>A</span>
          <span>S</span>
          <span>H</span>
        </Typography>
        <Typography variant='h2' className={styles.glowText}>
          <span>T</span>
          <span>R</span>
          <span>A</span>
          <span>C</span>
          <span>K</span>
          <span>E</span>
          <span>R</span>
        </Typography>
      </div>
      <Typography variant='subtitle1' classes={{ root: styles.subtitle }}>
        Track Your cash movement with ease
      </Typography>
      <svg
        className={styles.curve}
        version='1.1'
        viewBox='0 0 135.87 41.319'
        xmlns='http://www.w3.org/2000/svg'
      >
        <g transform='translate(5.1618 -101.68)'>
          <path d='m-5.1618 141.14c43.775-2.0845 72.303-10.174 91.574-20.944 29.649-16.57 47.402-20.394 49.127-17.747 0 0-2.9893 33.247-3.1169 43.982-8.9429 0.47794-56.289 4.3573-98.257 0.54109-6.2059-0.56431-39.326 4.7506-39.326-0.54109z' />
        </g>
      </svg>
    </section>
  );
};

export default LandingPage;
