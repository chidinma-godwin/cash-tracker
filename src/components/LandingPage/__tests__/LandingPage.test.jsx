import { fireEvent, render } from '@testing-library/react';

import { useRouter } from 'next/router';
import LandingPage from '..';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('LandingPage', () => {
  const mockRouterPush = jest.fn();
  beforeEach(() => {
    useRouter.mockImplementation(() => ({
      push: mockRouterPush,
    }));
  });
  it('should match snapshot', () => {
    const { asFragment } = render(<LandingPage />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('navigates to login page if "Login" button is clicked', () => {
    const { getByTestId } = render(<LandingPage />);

    fireEvent.click(getByTestId('login'));

    expect(mockRouterPush).toHaveBeenCalledWith(
      '/account/[location]',
      '/account/login'
    );
  });
  it('navigates to sign up page if "Sign Up" button is clicked', () => {
    const { getByTestId } = render(<LandingPage />);

    fireEvent.click(getByTestId('sign-up'));

    expect(mockRouterPush).toHaveBeenCalledWith(
      '/account/[location]',
      '/account/signup'
    );
  });
});
