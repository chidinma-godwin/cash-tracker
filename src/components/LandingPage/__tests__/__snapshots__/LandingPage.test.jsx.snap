// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-intro-1"
  >
    <div
      class="makeStyles-buttonWrapper-2"
    >
      <a
        class="makeStyles-buttonStyles-3"
        data-testid="login"
        role="button"
        tabindex="0"
      >
        <span />
        <span />
        <span />
        <span />
        Login
      </a>
      <a
        class="makeStyles-buttonStyles-3"
        data-testid="sign-up"
        role="button"
        tabindex="0"
      >
        <span />
        <span />
        <span />
        <span />
        Sign Up
      </a>
    </div>
    <div
      class="makeStyles-glowWrapper-5"
    >
      <h2
        class="MuiTypography-root makeStyles-glowText-6 MuiTypography-h2"
      >
        <span>
          C
        </span>
        <span>
          A
        </span>
        <span>
          S
        </span>
        <span>
          H
        </span>
      </h2>
      <h2
        class="MuiTypography-root makeStyles-glowText-6 MuiTypography-h2"
      >
        <span>
          T
        </span>
        <span>
          R
        </span>
        <span>
          A
        </span>
        <span>
          C
        </span>
        <span>
          K
        </span>
        <span>
          E
        </span>
        <span>
          R
        </span>
      </h2>
    </div>
    <h6
      class="MuiTypography-root makeStyles-subtitle-8 MuiTypography-subtitle1"
    >
      Track Your cash movement with ease
    </h6>
    <svg
      class="makeStyles-curve-7"
      version="1.1"
      viewBox="0 0 135.87 41.319"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        transform="translate(5.1618 -101.68)"
      >
        <path
          d="m-5.1618 141.14c43.775-2.0845 72.303-10.174 91.574-20.944 29.649-16.57 47.402-20.394 49.127-17.747 0 0-2.9893 33.247-3.1169 43.982-8.9429 0.47794-56.289 4.3573-98.257 0.54109-6.2059-0.56431-39.326 4.7506-39.326-0.54109z"
        />
      </g>
    </svg>
  </section>
</DocumentFragment>
`;
